app/src/components/MemberCard.vue

```diff
@@ -1,80 +1,66 @@
 <template>
-  <tr @click="showModal = true" class="cursor-pointer">
-    <th>{{ memberId }}</th>
-    <th>{{ memberName }}</th>
-    <th>{{ walletAddress }}</th>
+  <tr @click="emits('toggleUpdateMemberForm', member)" class="cursor-pointer">
+    <th>{{ member.id }}</th>
+    <th>{{ member.name }}</th>
+    <th>{{ member.walletAddress }}</th>
     <th>Action</th>
   </tr>
 
   <dialog
     id="my_modal_20"
+    v-if="showUpdateMemberModal"
     class="modal modal-bottom sm:modal-middle"
-    :class="{ 'modal-open': showModal }"
+    :class="{ 'modal-open': showUpdateMemberModal }"
   >
     <div class="modal-box">
       <button
         class="btn btn-sm btn-circle btn-ghost absolute right-2 top-2"
-        @click="showModal = !showModal"
+        @click="emits('toggleUpdateMemberForm', {})"
       >
-        ✕
+        <span>x</span>
       </button>
       <h1 class="font-bold text-2xl">Update Member Details</h1>
       <hr class="" />
-      <label class="input input-bordered flex items-center gap-2 input-md mt-2">
-        <input type="text" class="w-24" v-model="nameInput" />
+      <label
+        :class="{ 'input-error': !updateMemberInput.isValid }"
+        class="input input-bordered flex items-center gap-2 input-md mt-2"
+      >
+        <input type="text" class="w-24" v-model="updateMemberInput.name" />
         |
-        <input type="text" class="grow" v-model="walletInput" />
+        <input type="text" class="grow" v-model="updateMemberInput.walletAddress" />
       </label>
       <div class="flex mt-2 justify-between">
-        <button class="btn btn-error size-sm" @click="deleteMember">Delete</button>
-        <button class="btn btn-primary" @click="updateMember">Update</button>
+        <button class="btn btn-error size-sm" @click="emits('deleteMember', updateMemberInput.id)">
+          Delete
+        </button>
```

Comment:
and also we can refactor this button into single component, and the props will be the color of the button and the size and the function of the click

Reply:
I think daisy ui already simplifies this for us and the whole point of Tailwind and Daisy UI is to define class names on the go.
A generic component, I feel, would take away the point of it.